tests/test_storageclass_deletion.py

```diff
@@ -0,0 +1,58 @@
+"""
+Automates the test OCS-297:-
+1) Create a Storage Class
+2) Create a PVC
+3) Delete the corresponding storage class
+"""
+import logging
+import pytest
+from ocsci.testlib import tier1, ManageTest
+from tests import helpers
+from ocs_ci.ocs import constants
+
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope='class')
+def test_fixture(request):
+    """
+    Deleting a Storage class
+    """
+    global SECRET, PVC, STORAGE_CLASS
+    log.info("Creating RBD Secret")
+    SECRET = helpers.create_secret(constants.CEPHBLOCKPOOL)
+
+    log.info("Creating RBD StorageClass")
+    STORAGE_CLASS = helpers.create_storage_class(
+        constants.CEPHBLOCKPOOL, 'rbd', SECRET.name
+    )
+
+    def finalizer():
+        teardown()
+    request.addfinalizer(finalizer)
+
+
+def teardown():
+    """
+    Remove the resources after execution of tests
+    """
+    log.info("Deleting PVC")
+    assert PVC.delete()
```

Comment:
you shouldn't use asserts in finalizers as if you will throw assertion exception here the second resource in your case SECRET won't . be deleted